client/python/tests/integration/test_no_auth.py

```diff
@@ -1,113 +1,167 @@
-import time
 import uuid
-
+import time
+import os
 import grpc
+
+import pytest
+
+
 from armada_client.client import ArmadaClient
 from armada_client.k8s.io.api.core.v1 import generated_pb2 as core_v1
 from armada_client.k8s.io.apimachinery.pkg.api.resource import (
     generated_pb2 as api_resource,
 )
 
-no_auth_client = ArmadaClient(channel=grpc.insecure_channel(target="127.0.0.1:50051"))
 
+def client() -> ArmadaClient:
+    server_name = os.environ.get("ARMADA_SERVER", "localhost")
+    server_port = os.environ.get("ARMADA_PORT", "50051")
+    server_ssl = os.environ.get("ARMADA_SSL", "false")
 
-def test_submit_job():
-    queue_name = f"queue-{uuid.uuid1()}"
-    job_set_name = f"set-{uuid.uuid1()}"
-    no_auth_client.create_queue(name=queue_name, priority_factor=200)
-    no_auth_client.submit_jobs(
-        queue=queue_name, job_set_id=job_set_name, job_request_items=submit_sleep_job()
-    )
-    # Jobs can must be finished before deleting queue
-    no_auth_client.cancel_jobs(queue=queue_name, job_set_id=job_set_name)
-    no_auth_client.delete_queue(name=queue_name)
-    time.sleep(1)
+    if server_ssl == "true":
```

Comment:
nit: Would be nice if we made server_ssl a real bool instead of this.

```
>>> import distutils
>>> distutils.util.strtobool("falsey")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.8/distutils/util.py", line 319, in strtobool
    raise ValueError("invalid truth value %r" % (val,))
ValueError: invalid truth value 'falsey'
>>> distutils.util.strtobool("truthy")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.8/distutils/util.py", line 319, in strtobool
    raise ValueError("invalid truth value %r" % (val,))
ValueError: invalid truth value 'truthy'
>>> distutils.util.strtobool("true")
1
>>> distutils.util.strtobool("0")
0
>>> distutils.util.strtobool("false")
0
>>> distutils.util.strtobool("1")
1
```

Reply:
Enums can solve this, I shall do that.